compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -37,13 +45,18 @@ impl<'a> Lexer<'a> {
         (Tokens(tokens), errors)
     }
 
-    pub fn new(source: &'a str) -> Self {
+    pub fn single_source(source: &'a str) -> Self {
+        Self::new(SrcId::default(), source)
+    }
```

Comment:
```suggestion
    #[cfg(test)]
    pub fn from_raw_string(source: &'a str) -> Self {
        Self::new(SrcId::default(), source)
    }
```
I wasn't sure on the `single_source` name either

Reply:
renamed to `from_string`